examples/text-classification/new_tf_text_classification.py

```diff
@@ -0,0 +1,498 @@
+
+    if data_args.train_file.endswith(".csv"):
+        # Loading a dataset from local csv files
+        datasets = load_dataset("csv", data_files=data_files, cache_dir=model_args.cache_dir)
+    else:
+        # Loading a dataset from local json files
+        datasets = load_dataset("json", data_files=data_files, cache_dir=model_args.cache_dir)
+    # See more about loading any type of standard or custom dataset at
+    # https://huggingface.co/docs/datasets/loading_datasets.html.
+    # endregion
+
+    # region Label preprocessing
+    # Trying to have good defaults here, don't hesitate to tweak to your needs.
+    is_regression = datasets["train"].features["label"].dtype in ["float32", "float64"]
+    if is_regression:
+        num_labels = 1
+    else:
+        # A useful fast method:
+        # https://huggingface.co/docs/datasets/package_reference/main_classes.html#datasets.Dataset.unique
+        label_list = datasets["train"].unique("label")
+        label_list.sort()  # Let's sort it for determinism
+        num_labels = len(label_list)
+    # endregion
+
+    # region Load pretrained model and tokenizer
+    # Set seed before initializing model
+    set_seed(training_args.seed)
+    #
+    # In distributed training, the .from_pretrained methods guarantee that only one local process can concurrently
+    # download model & vocab.
+    config = AutoConfig.from_pretrained(
+        model_args.config_name if model_args.config_name else model_args.model_name_or_path,
+        num_labels=num_labels,
+        cache_dir=model_args.cache_dir,
+        revision=model_args.model_revision,
+        use_auth_token=True if model_args.use_auth_token else None,
+    )
+    tokenizer = AutoTokenizer.from_pretrained(
+        model_args.tokenizer_name if model_args.tokenizer_name else model_args.model_name_or_path,
+        cache_dir=model_args.cache_dir,
+        use_fast=model_args.use_fast_tokenizer,
+        revision=model_args.model_revision,
+        use_auth_token=True if model_args.use_auth_token else None,
+    )
+    if checkpoint is None:
+        model_path = model_args.model_name_or_path
+    else:
+        model_path = checkpoint
+    model = TFAutoModelForSequenceClassification.from_pretrained(
+        model_path,
+        config=config,
+        cache_dir=model_args.cache_dir,
+        revision=model_args.model_revision,
+        use_auth_token=True if model_args.use_auth_token else None,
+    )
+    optimizer = tf.keras.optimizers.Adam(
+        learning_rate=training_args.learning_rate,
+        beta_1=training_args.adam_beta1,
+        beta_2=training_args.adam_beta2,
+        epsilon=training_args.adam_epsilon,
+        clipnorm=training_args.max_grad_norm)
+    if is_regression:
+        loss = tf.keras.losses.MeanSquaredError()
+    else:
+        loss = tf.keras.losses.SparseCategoricalCrossentropy(from_logits=True)
+    model.compile(optimizer=optimizer, loss=loss)
```

Comment:
I wouldn't put this in `# region Load pretrained model and tokenizer`, but maybe something like `# region Define optimizer, loss and compilation`? I feel like this is something that should be separate

Reply:
Done!